### ISSUE TITLE
@babel/runtime ^7.2.0 breaks reactstrap because installs 7.3.0

### ISSUE THREAD

**Issue by stahlmanDesign** [reactstrap/reactstrap#1373]
- components: `NavLink NavBar`
- reactstrap version `#7.1.0`
- import method `umd/csj/es` 
- react version `#16.7.0`
- bootstrap version `#4.2.1`

### What is happening?

@babel/runtime updated to 7.3.0 and it is a breaking change for reactstrap

reactstrap has dependency `@babel/runtime: ^7.2.0`

This means it installs 7.3.0 and that breaks reactstrap building
Suggested quick fix for reactstrap - change to tilde(~) `@babel/runtime: ~7.2.0` so it only updates 7.2.x

### What should be happening?
should build

### Steps to reproduce issue

1. include reactstrap in a project like create-react-app and build

### Error message in console

```
TypeError: Cannot convert undefined or null to object
    at Function.getOwnPropertyDescriptors (<anonymous>)
    at _objectSpread (/app/node_modules/reactstrap/node_modules/@babel/runtime/helpers/objectSpread.js:19:46)
    at /app/node_modules/reactstrap/lib/Collapse.js:146:43
    at Transition.render (/app/node_modules/react-transition-group/Transition.js:418:14)
    at c (/app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:30:231)
    at Ga (/app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:31:1)
    at a.render (/app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:36:367)
    at a.read (/app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:35:450)
    at renderToString (/app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:47:82)
    at renderPage (/app/node_modules/next/dist/server/render.js:319:26)
```

NOTE: if previously installed, delete your package-lock.json to reproduce bug. Packge-lock.json was making my build work locally but not when deployed, which was fresh and not respecting package-lock

**Comment by stahlmanDesign** [comment]
We fixed this by modifying the ^ with a ~ in package-lock.json

![capture d ecran le 2019-01-21 a 19 12 42](https://user-images.githubusercontent.com/972804/51504746-9c957180-1db0-11e9-81c3-2554edfd8706.png)

**Comment by chulphan** [comment]
@stahlmanDesign I have same issue. Did you solve the issue after fix ^ to ~ in node_modules/reactstrap/package.json??

**Comment by emurray1** [comment]
The above fix resolved the bug for me as well.  Important to note you will most likely need to delete your package-lock.json and reinstall.

**Comment by chulphan** [comment]
@stahlmanDesign @emurray1 Thank you for your answers. ^^

**Comment by chulphan** [comment]
@stahlmanDesign I modified ^ to ~ in package-lock.json in rootdir and then tried deploy but the issue cannot resolve..

**Comment by chulphan** [comment]
@stahlmanDesign I tried npm ci to install dependencies, the issue resolved!! thank you!!!^^

**Comment by TheSharpieOne** [comment]
This was a babel issue and babel fixed it in [7.3.1](https://github.com/babel/babel/releases/tag/v7.3.1)
https://github.com/babel/babel/pull/9379

If anyone manually changed `package-lock.json` files, I suggest you change them back.

Also, to ensure CI builds use the same lib versions you use locally either use `package-lock.json` or `yarn.lock` and check those files in. Use the same tool (npm vs yarn) in your build environment as you do locally. For instanace ,using `npm ci` makes CI builds faster by performing a clean install with the specific versions specified in the `package-lock.json`. Even if a range is specified, the package-lock or yarn.lock will lock the version to a specific version.
